kpatch-build/kpatch-build

```diff
@@ -1107,6 +1142,7 @@ declare -a MAKEVARS
 if [[ -n "$CONFIG_CC_IS_CLANG" ]]; then
 	MAKEVARS+=("CC=${KPATCH_CC_PREFIX}${CLANG}")
 	MAKEVARS+=("HOSTCC=clang")
+	MAKEVARS+=("LLVM=1")
```

Comment:
This is only a one-line change but it has a big impact and belongs in a separate commit with proper justification.  Also I wonder how it interacts with the CONFIG_LD_IS_LLD check.

Reply:
It appears to me that I need to pass in LLVM=1 to the make command. Otherwise, the make will fail. I have the same change for PGO support.

Reply:
It is possible for the user to specify CC=clang and LD=ld.lld separately. And LLVM=1 will enable a set of utilities. From llvm.rst:
```
LLVM has substitutes for GNU binutils utilities. They can be enabled individually.
The full list of supported make variables::
        make CC=clang LD=ld.lld AR=llvm-ar NM=llvm-nm STRIP=llvm-strip \
          OBJCOPY=llvm-objcopy OBJDUMP=llvm-objdump READELF=llvm-readelf \
          HOSTCC=clang HOSTCXX=clang++ HOSTAR=llvm-ar HOSTLD=ld.lld
To simplify the above command, Kbuild supports the ``LLVM`` variable::
        make LLVM=1
```
I changed the code to use CONFIG_AS_IS_LLVM to detect the use of LLVM=1. I think this should cover most use cases.